ddtrace/tracer/option.go

```diff
@@ -186,16 +187,23 @@ func WithPropagator(p Propagator) StartOption {
 	}
 }
 
-// WithServiceName sets the default service name to be used with the tracer. It is
-// deprecated in favour of WithService and will be removed in the next major version.
+// WithServiceName sets the default service name to be used with the tracer. This will
+// not affect the service name used by any integrations. It is deprecated in favour of
+// WithService and will be removed in the next major version.
 func WithServiceName(name string) StartOption {
-	return WithService(name)
+	return func(c *config) {
+		c.serviceName = name
+		globalconfig.SetServiceName("")
+	}
 }
 
-// WithService sets the default service name to be used with the tracer.
+// WithService sets the default service name to be used with the tracer. It will also be
+// used as the service name for server and framework integrations
+// (e.g. http servers, grpc, etc.)
```

Comment:
Since we consider this behaviour normal and as it should be - would it be better to just:

```suggestion
// WithService sets the default service name for the program.
```

Reply:
Agreed. I went back and forth on this, but I'm ok being more terse.